pkg/subscription/clientid/clientid.go

```diff
@@ -0,0 +1,161 @@
+package clientid
+
+import (
+	"fmt"
+
+	"github.com/Axway/agent-sdk/pkg/cache"
+
+	"github.com/Axway/agent-sdk/pkg/apic"
+	"github.com/Axway/agents-mulesoft/pkg/anypoint"
+	"github.com/Axway/agents-mulesoft/pkg/subscription"
+	"github.com/sirupsen/logrus"
+)
+
+type clientId struct {
+	apc anypoint.Client
+}
+
+const (
+	name = "client-id-enforcement"
+)
+
+func init() {
+	subscription.Register(func(apc anypoint.Client) subscription.Handler {
```

Comment:
I think this would be more clear if this was not done in the init func. subscription manager could be updated to take these callback func and pass them to Register.